src/Stepper/Step.js

```diff
@@ -0,0 +1,95 @@
+import React, {Component, PropTypes} from 'react';
+
+const getStyles = ({index}, {stepper}) => {
+  const {orientation} = stepper;
+  const styles = {
+    root: {
+      flex: '0 0 auto',
+    },
+  };
+
+  if (index > 0) {
+    if (orientation === 'horizontal') {
+      styles.root.marginLeft = -6;
+    } else if (orientation === 'vertical') {
+      styles.root.marginTop = -14;
+    }
+  }
+
+  return styles;
+};
+
+export default class Step extends Component {
+
+  static propTypes = {
+    /**
+     * Sets the step as active. Is passed to child components.
+     */
+    active: PropTypes.bool,
+    /**
+     * Should be `Step` sub-components such as `StepLabel`.
+     */
+    children: PropTypes.node,
+    /**
+     * Mark the step as completed. Is passed to child components.
+     */
+    completed: PropTypes.bool,
+    /**
+     * Mark the step as disabled, will also disable the button if
+     * `StepButton` is a child of `Step`. Is passed to child components.
+     */
+    disabled: PropTypes.bool,
+    /**
+     * @ignore
+     * Used internally for numbering.
+     */
+    index: PropTypes.number,
+    /**
+     * @ignore
+     */
+    last: PropTypes.bool,
+    /**
+     * Override the inline-style of the root element.
+     */
+    style: PropTypes.object,
+  };
+
+  static contextTypes = {
+    muiTheme: PropTypes.object.isRequired,
+    stepper: PropTypes.object,
+  };
+
+  renderChild = (child) => {
+    const {
+      active,
+      completed,
+      disabled,
+      index,
+      last,
+    } = this.props;
+
+    const icon = index + 1;
+
+    return React.cloneElement(child, Object.assign(
+      {active, completed, disabled, icon, last},
```

Comment:
Personally, I'm a fan of this short-hand syntax for creating objects 😄. Though I think we may have discussed in another issue we may consider setting up an eslint rule to disallow this.

@oliviertassinari Is this fine?